crates/bevy_pbr/src/light.rs

```diff
@@ -280,6 +279,12 @@ impl Default for AmbientLight {
     }
 }
 
+pub fn add_ambient_lights(mut commands: Commands, cameras: Query<Entity, With<Camera>>) {
+    for entity in &cameras {
+        commands.entity(entity).insert(AmbientLight::default());
+    }
+}
```

Comment:
This would add the component every frame to every cameras, overwriting the previous value. This would prevent users from setting their own ambient light. You shouldn't need such system. Instead, it's likely the best solution is to add the `AmbientLight` component to `Camera3dBundle`.

Reply:
Thanks for taking the time to give feedback!

Yeah I kind of arrived at the same conclusion but I am now facing a different issue. Currently `AmbientLight` is part of the pbr crate/feature and is only ever used there. Does this mean it should be behind a `#[cfg(feature = "bevy_pbr")]` directive?:

```rust
#[derive(Bundle)]
pub struct Camera3dBundle {
    pub camera: Camera,
    pub camera_render_graph: CameraRenderGraph,
    pub projection: Projection,
    pub visible_entities: VisibleEntities,
    pub frustum: Frustum,
    pub transform: Transform,
    pub global_transform: GlobalTransform,
    pub camera_3d: Camera3d,
    pub tonemapping: Tonemapping,
    #[cfg(feature = "bevy_pbr")]
    pub ambient_light: AmbientLight,
}
```

Or do we just move the whole struct into the `bevy_core_pipeline` crate? Are there other options I'm overlooking?

Reply:
yeah, this is a tough decision, because ambient light relates to pbr while core_pipeline is just the generic bevy rendering infra (no expert actually, might be wrong).

An alternative would be to not add `AmbientLight` to `Camera3dBundle` and assume default  when the component is not present on the camera entity. (that is, in the `prepare_lights` system, you'd accept a `Option<&ExtractedAmbientLight>` instead of `&ExtractedAmbientLight`, and go with the default value when `None`) It's easy for the user to do:
```rust
commands.spawn((
  Camera3dBundle {
    //...
  }),
  AmbientLight { … },
))
```
If they want a custom ambient light.

The cfg feature would require `bevy_core_pipeline` to depend on `bevy_pbr` which is not OK IMO because of circular dependency (even on feature gate). I would really prefer if this was a field of Camera3dBundle, but it seems not possible. Moving the struct to `bevy_core_pipeline` seems fine, but it would be misleading (ie: it does nothing when not using bevy_pbr). Moving the struct to bevy_core_pipeline and toggling it on feature flag seems overly complex.

Reply:
> An alternative would be to not add AmbientLight to Camera3dBundle and assume default when the component is not present on the camera entity. (that is, in the prepare_lights system, you'd accept a Option<&ExtractedAmbientLight> instead of &ExtractedAmbientLight, and go with the default value when None)

This is the approach I'm currently taking, but it's definitely harder to discover as a user. I will finish the implementation with this direction and maybe then a SME (have to look up who) can chime in. This seems relevant because #7194 will probably deal with a similar issue. So will any another (future) per-camera feature.

Reply:
I agree, it makes it not discoverable at all, that's why I'd prefer it to be a field of Camera3dBundle. Looks like someone already did pick up #7194 with: #7215 `bevy_core_pipeline` depends on `bevy_render`, which is what defines `Msaa`, so it doesn't have the same issue.

Reply:
If we want to go forward with this, I think it should behave like [`UiCameraConfig`](https://docs.rs/bevy/0.9.1/bevy/prelude/struct.UiCameraConfig.html). Default behaviour if not present, custom if present